lib/ansiblelint/utils.py

```diff
@@ -258,18 +247,29 @@ def _taskshandlers_children(basedir, k, v, parent_type):
     return results
 
 
-def append_children(taskhandler, basedir, k, parent_type, results):
-    # when taskshandlers_children is called for playbooks, the
-    # actual type of the included tasks is the section containing the
-    # include, e.g. tasks, pre_tasks, or handlers.
-    if parent_type == 'playbook':
-        playbook_section = k
-    else:
-        playbook_section = parent_type
-    results.append({
-        'path': path_dwim(basedir, taskhandler),
-        'type': playbook_section
-    })
+def _maybe_th_children_for_tasks_or_playbooks(th, basedir, k, parent_type):
+    """Try to get children of taskhandler for include/import tasks/playbooks."""
+    for tht in ('include', 'include_tasks', 'import_playbook', 'import_tasks'):
+        if tht in th:
+            return {
+                'path': path_dwim(basedir, th[tht]),
+                'type': k if parent_type == 'playbook' else parent_type
+            }
+
+    return None
+
+
+def _validate_th_action_for_role(th_action):
+    name = th_action.get("name", None)
```

Comment:
This changes the semantics of this check creating the ambiguity when the key exists but is `None`. Refactoring shouldn't influence the original logic.